crates/task-impls/src/consensus/helpers.rs

```diff
@@ -511,7 +491,7 @@ async fn publish_proposal_from_commitment_and_metadata<TYPES: NodeType>(
     // FIXME - This is not great, and will be fixed later.
     // If it's > July, 2024 and this is still here, something has gone horribly wrong.
     let cnm = commitment_and_metadata
-        .take()
+        .clone()
```

Comment:
If we're cloning, and we don't have this as mutable anymore, we might be able to blow away the FIXME here. The issue that the FIXME was added to alert to was that this whole process was super obtuse due to needing to maintain backwards compatibility with the state handling in the consensus task.

Reply:
I also think the original backward state handling needs some rework, but I'd like to be more certain if we can make `commitment_and_metadata` immutable and replace `take` with `clone` here.

We probably can, since it looks like we don't have a logic where we do an operation depending on whether `commitment_and_metadata` is `Some`, except for setting it to `None` which is fine. If so, we can simplify the mutable handling in some other places as well (not necessarily in this PR), but I just want to make sure this is the case and that we are not skipping the state update unintentionally.

Reply:
Hopefully we're going to merge the new tasks soon and this all becomes a distant memory.

Reply:
We can't remove this in the current code just yet, if that's what you are suggesting.  We should be able to get rid of this with the dependency tasks.